tests/testthat/test-compare.R

```diff
@@ -115,13 +106,19 @@ test_that("xtfrm.vctrs_vctr works for variety of base classes", {
   expect_equal(xtfrm.vctrs_vctr(letters[x]), x)
 })
 
-test_that("vec_proxy_compare() refuses to deal with lists", {
-  expect_error(vec_proxy_compare(list()), class = "vctrs_error_unsupported")
+test_that("vec_proxy_order() orders list using order of appearance", {
+  x <- 1:2
+  y <- 2:4
+  z <- 3
```

Comment:
Might make sense to do `z <- "a"` just to emphasise that these aren't the same type.